Composer/packages/test-utils/src/mergeConfig.ts

```diff
@@ -4,13 +4,14 @@
 
 import mergeWith from 'lodash/mergeWith';
 import isArray from 'lodash/isArray';
+import cloneDeep from 'lodash/cloneDeep';
 
 function mergeConfig<T extends object>(base: T, overrides?: T) {
   if (!overrides) {
     return base;
   }
 
-  return mergeWith(base, overrides, (objValue, srcValue) => {
+  return mergeWith(cloneDeep(base), overrides, (objValue, srcValue) => {
```

Comment:
why the need to cloneDeep here?

Reply:
`mergeWith(base, extra)` will manipulate the base object rather than create a new one, eventually `baseConfig === nodeConfig === reactConfig`. It causes the baseConfig be merged twice and nodeConfig is poluted by reactConfig.

It's a similar problem to Object.assgin. `Object.assign(base, a, b, ...)` vs `Object.assign({}, base, a, b, ...)`